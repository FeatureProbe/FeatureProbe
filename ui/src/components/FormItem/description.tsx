import { SyntheticEvent } from 'react';
import { Form, FormTextAreaProps, TextAreaProps } from 'semantic-ui-react';
import { FormattedMessage, useIntl } from 'react-intl';
import styles from './index.module.scss';

interface IProps extends FormTextAreaProps {
  value: string;
  size?: 'big' | 'small' | 'mini' | 'large' | 'huge' | 'massive' | undefined;
  className?: string;
  onChange(e: SyntheticEvent, detail: TextAreaProps): void;
}

const FormItemDescription = (props: IProps) => {
  const intl = useIntl();
  const { value, className, onChange } = props;

  return (
    <div className={className}>
      <Form.Field>
        <label>
          <FormattedMessage id='common.description.text' />
        </label>
        <Form.TextArea 
          value={value} 
          placeholder={intl.formatMessage({id: 'common.description.required'})}
          className={styles.input}
          onChange={onChange}
        />
      </Form.Field>
    </div>
  );
};

export default FormItemDescription;