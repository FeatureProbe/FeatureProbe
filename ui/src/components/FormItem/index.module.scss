.input {
  // width: 500px !important;
  font-size: 13px;
}

.label-required {
  color: #ff5f57;
  font-size: 20px;
  vertical-align: middle;
  margin-right: 4px;
  height: 16px;
  display: inline-block;
}

.error-text {
  color: #f46a6a;
  margin-top: -15px;
  margin-bottom: 15px;
  font-size: 12px;
  transform: scale(0.83);
  transform-origin: left;
}


.tip-text {
  color: #74788d;
  margin-top: -15px;
  margin-bottom: 15px;
  font-size: 12px;
  transform: scale(0.83);
  transform-origin: left center;
}

.icon-question {
  vertical-align: middle;
  margin-left: 4px;
  color: #74788d;
}
