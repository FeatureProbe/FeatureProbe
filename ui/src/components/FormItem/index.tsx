import { CSSProperties, ReactNode } from 'react';
import { FieldErrors } from 'react-hook-form';
import { Form } from 'semantic-ui-react';
import 'index.scss';

interface FormItemProps {
  label: ReactNode;
  required?: boolean;
  error?: FieldErrors[string];
  errorCss?: CSSProperties
  className?: string;
}

const FormItem: React.FC<FormItemProps> = (props) => {
  const { label, children, required, error, errorCss, className } = props;

  return (
    <Form.Field className={className}>
      <label>
        {required && <span className='label-required'>*</span>}
        {label}
      </label>
      {children}
      {error && <div style={errorCss} className='error-text'>{error.message}</div>}
    </Form.Field>
  );
};

export default FormItem;
