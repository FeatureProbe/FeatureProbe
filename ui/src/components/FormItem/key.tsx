import { SyntheticEvent, useEffect } from 'react';
import { Form, InputOnChangeData, Popup } from 'semantic-ui-react';
import { FormattedMessage, useIntl } from 'react-intl';
import { UseFormRegister, FieldValues, FieldErrors } from 'react-hook-form';
import Icon from 'components/Icon';
import styles from './index.module.scss';

interface IProps {
  value: string;
  disabled: boolean;
  errors: FieldErrors;
  showPopup?: boolean;
  popupText?: string;
  className?: string;
  size?: 'big' | 'small' | 'mini' | 'large' | 'huge' | 'massive' | undefined;
  register: UseFormRegister<FieldValues>;
  onChange(e: SyntheticEvent, detail: InputOnChangeData): void;
}

const FormItemKey = (props: IProps) => {
  const intl = useIntl();
  const { value, errors, disabled, showPopup, popupText, size, className, register, onChange } = props;

  useEffect(() => {
    register('key', { 
      required: intl.formatMessage({id: 'common.key.required'}),
      minLength: {
        value: 2,
        message: intl.formatMessage({id: 'common.minimum.two'})
      },
      maxLength: {
        value: 30,
        message: intl.formatMessage({id: 'common.maximum.thirty'})
      },
      pattern: {
        value: /^[A-Z0-9._-]+$/i,
        message: intl.formatMessage({id: 'common.key.invalid'})
      }
    });
  }, [intl, register]);

  return (
    <div className={className}>
       <Form.Field>
        <label>
          <span className={styles['label-required']}>*</span>
          <FormattedMessage id='common.key.text' />
          {
            showPopup && (
              <Popup
                inverted
                className='popup-override'
                trigger={
                  <Icon customclass={styles['icon-question']} type='question' />
                }
                content={popupText}
                position='top center'
              />
            )
          }
        </label>

        <Form.Input
          name='key'
          size={size}
          className={styles.input}
          value={value}
          placeholder={intl.formatMessage({id: 'common.key.required'})}
          error={ errors.key ? true : false }
          disabled={disabled}
          onChange={onChange}
        />
      </Form.Field>
      { errors.key && <div className={styles['error-text']}>{ errors.key.message }</div> }
      <div className={styles['tip-text']}>
        <FormattedMessage id='common.key.tips' />
      </div>
    </div>
  );
};

export default FormItemKey;