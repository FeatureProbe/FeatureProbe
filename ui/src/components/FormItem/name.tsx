import { ReactNode, SyntheticEvent, useEffect } from 'react';
import { Form, FormInputProps, InputOnChangeData } from 'semantic-ui-react';
import { FormattedMessage, useIntl } from 'react-intl';
import { UseFormRegister, FieldValues, FieldErrors } from 'react-hook-form';
import styles from './index.module.scss';

interface IProps extends FormInputProps {
  value: string;
  errors: FieldErrors;
  size?: 'big' | 'small' | 'mini' | 'large' | 'huge' | 'massive' | undefined;
  className?: string;
  register: UseFormRegister<FieldValues>;
  onChange(e: SyntheticEvent, detail: InputOnChangeData): void;
  labelRender?: ReactNode;
}

const FormItemName = (props: IProps) => {
  const intl = useIntl();
  const { value, errors, size, className, register, onChange, labelRender } = props;

  useEffect(() => {
    register('name', { 
      required: intl.formatMessage({id: 'common.name.required'}),
    });
  }, [intl, register]);

  return (
    <div className={className}>
      <Form.Field>
        <label>
          <span className={styles['label-required']}>*</span>
          {labelRender ?? <FormattedMessage id='common.name.text' />}
        </label>
        <Form.Input
          name='name'
          size={size}
          className={styles.input}
          value={value}
          placeholder={intl.formatMessage({id: 'common.name.required'})}
          error={ errors.name ? true : false }
          onChange={onChange}
        />
      </Form.Field>
      { errors.name && <div className={styles['error-text']}>{ errors.name.message }</div> }
    </div>
  );
};

export default FormItemName;