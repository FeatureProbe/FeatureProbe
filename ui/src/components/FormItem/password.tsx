import { SyntheticEvent, useEffect } from 'react';
import { Form, InputOnChangeData } from 'semantic-ui-react';
import { FormattedMessage, useIntl } from 'react-intl';
import { UseFormRegister, FieldValues, FieldErrors } from 'react-hook-form';
import styles from './index.module.scss';

interface IProps {
  errors: FieldErrors;
  register: UseFormRegister<FieldValues>;
  onChange(e: SyntheticEvent, detail: InputOnChangeData): void;
}

const FormItemPassword = (props: IProps) => {
  const intl = useIntl();
  const { errors, register, onChange } = props;

  useEffect(() => {
    register('password', { 
      required: intl.formatMessage({id: 'login.password.required'}),
      minLength: {
        value: 4,
        message: intl.formatMessage({id: 'login.password.minlength'}),
      },
      maxLength: {
        value: 20,
        message: intl.formatMessage({id: 'login.password.maxlength'})
      },
      pattern: {
        value: /^[A-Z0-9._-]+$/i,
        message: intl.formatMessage({id: 'login.password.invalid'})
      }
    });
  }, [intl, register]);

  return (
    <>
      <Form.Field>
        <label>
          <span className={styles['label-required']}>*</span>
          <FormattedMessage id='common.password.text' />
        </label>

        <Form.Input
          name='password'
          className={styles.input}
          type='password'
          placeholder={intl.formatMessage({id: 'login.password.required'})} 
          error={ errors.password ? true : false }
          onChange={onChange}
        />
      </Form.Field>
      { errors.password && <div className={styles['error-text']}>{ errors.password.message }</div> }
      <div className={styles['tip-text']}>
        <FormattedMessage id='login.password.tips' />
      </div>
    </>
  );
};

export default FormItemPassword;