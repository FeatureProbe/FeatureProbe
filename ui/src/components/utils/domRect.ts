export class DOMRect {
    x: number;
    width: number;
    y: number;
    height: number;
    constructor(x: number, y: number, width: number, height: number) {
        this.x = x;
        this.y = y;
        this.width = width;
        this.height = height;
    }
}