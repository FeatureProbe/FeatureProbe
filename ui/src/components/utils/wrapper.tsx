import en_US from '../../locales/en-US.json';
import { IntlProvider } from 'react-intl';

export const IntlWrapper: React.FC = ({ children }) => {
  return (
    <IntlProvider locale="en" messages={en_US}>{children}</IntlProvider>
  );
};
