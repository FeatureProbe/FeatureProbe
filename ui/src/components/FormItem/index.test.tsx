import { render } from '@testing-library/react';
import '@testing-library/jest-dom';
import { FieldErrorsImpl, useForm } from 'react-hook-form';
import { IntlWrapper } from 'components/utils/wrapper';
import FormItem from '.';

it('Key snapshot', () => {
  (async () => {
    const { asFragment, rerender } = render(<FormItem label={'test label'}>test</FormItem>, {
      wrapper: IntlWrapper,
    });

    expect(asFragment()).toMatchSnapshot();

    rerender(
      <FormItem
        error={{
          key: {
            message: 'error message',
            type: 'deps',
          },
        }}
        label={'test label'}
        errorCss={{
          color: 'pink'
        }}
        required={true}
      />
    );

    expect(asFragment()).toMatchSnapshot();
  })();
});
