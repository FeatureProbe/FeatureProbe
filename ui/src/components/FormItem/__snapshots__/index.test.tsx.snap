// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Key snapshot 1`] = `
<DocumentFragment>
  <div
    class="field"
  >
    <label>
      test label
    </label>
    test
  </div>
</DocumentFragment>
`;

exports[`Key snapshot 2`] = `
<DocumentFragment>
  <div
    class="field"
  >
    <label>
      <span
        class="label-required"
      >
        *
      </span>
      test label
    </label>
    <div
      class="error-text"
      style="color: pink;"
    />
  </div>
</DocumentFragment>
`;
